tests/python_package_test/test_arrow.py

```diff
@@ -432,3 +432,15 @@ def test_predict_ranking():
         num_boost_round=5,
     )
     assert_equal_predict_arrow_pandas(booster, data)
+
+
+def test_arrow_categorical():
+    data = generate_random_arrow_table(10, 10000, 42)
```

Comment:
```suggestion
    data = generate_random_arrow_table(
        num_columns=3,
        num_datapoints=1_000,
        seed=42
    )
```

Since this is just about column names and not related to data shape, can we please use a smaller dataset? To save some time and memory in tests.